src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -374,6 +374,18 @@ internal BoundExpression BindToNaturalType(BoundExpression expression, BindingDi
                         result = BindObjectCreationForErrorRecovery(expr, diagnostics);
                     }
                     break;
+                case BoundUnconvertedInterpolatedString unconvertedInterpolatedString:
+                    {
+                        // We determine the best method of emitting as a string (either via Concat, Format, or the builder pattern)
+                        // during lowering, and it's not part of the publicly-visible API, unlike conversion to a builder type.
+                        result = new BoundInterpolatedString(
```

Comment:
I'm not sure I follow this comment. If we do building in lowering, why do we need to have this split here at all? Or am I confusing builder pattern and builder type as the same thing?

My assumption is that we'll add code here in  future PR to create some new kind of `BoundBuilderInterpoltedString` if we detect its actually a builder pattern/type one right?

Reply:
The code won't be here. The code will be when the interpolated string is converted to a non-`String` type.

Reply:
Hmm, I guess I'm missing something. Why do we need the `BoundUnconvertedInterpolatedString` / `BoundInterpolatedString` if we're converting to another type later on?

Reply:
> Why do we need the BoundUnconvertedInterpolatedString / BoundInterpolatedString if we're converting to another type later on?

We need it because we want to tell the difference between a string that has been converted and one that has not been converted. This is the same approach we take with other target-typed constructs such as switch expressions and ternaries. Happy to discuss over teams in more depth if you want.

Reply:
Ahh, got it, thanks.